### PR TITLE
Added convenience enums of fg and bg colors

### PR DESCRIPTION
**Pull request by tleonhardt**
Added convenience `ansi.fg` and `ansi.bg` enums of foreground and background colors which `ansi.style()` can now optionally use for its `fg` and `bg` arguments.  

This is to make it easier to autocomplete color names in an IDE.

NOTE: This does not break backward compatibility in any way - supplying a `str` color still works.

### CODE REVIEW

File: cmd2/ansi.py [hunk 1]
```diff
@@ -25,57 +26,91 @@
...
+        """Make the value the string representation instead of the enum name."""
+        return self.value
+
+    @staticmethod
+    def colors() -> List[str]:
+        """Return a list of color names."""
+        return [color.name for color in fg]
+
+    @staticmethod
+    def get_value(color: str) -> str:
```
[the remark this answers is not in the document]
Comment (tleonhardt) [reply cmd2/ansi.py]: I changed it to `name`

**Comment by kotfu** [comment]
Actually, this does break backward compatibility. e66dc6f7078e1d4928ef58ccf48f4718b61d8c35 removes `FG_COLORS` and 'BG_COLORS' from ansi.py. Any application using those constants will no longer run.

**Comment by kotfu** [comment]
And the changelog doesn't say anything about the removal of those constants either.

**Comment by tleonhardt** [comment]
@kotfu Good call!  I will update the changelog one way or another!

If you are OK with the removal I will simply document it.

If not, I will mark them as deprecated and generate them from the enums.

**Comment by kmvanbrunt** [comment]
I assume most people just passed the string names. Probably won't matter too much removing those dictionaries.

**Comment by kotfu** [comment]
I'm OK with the removal as long as it goes in the changelog, but we are going to have to be much more thoughtful going forward about our process for change. If we had already cut a 1.0.0 release and were strictly following semantic versioning, these constants would have to be marked as deprecated in a 1.1.0 release, and then removed in a 2.0.0 release.

**Pull request merged**